### PR TITLE
[Photos] Save full resolution when capturing image via camera

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Fixes https://github.com/google/ground-platform/issues/761

Things done:
 * Save photos captured / selected from gallery to external storage directory
 * Move photo handling related code to UserMediaRepository
 * Remove bitmap/photo handling code from `FileUtil` and let it be generic utility class.
 * Cleanup EditObservationViewModel by moving unrelated code to media related classes.

@gino-m  PTAL?

### CODE REVIEW

File: gnd/src/main/java/com/google/android/gnd/repository/UserMediaRepository.java [hunk 1]
```diff
@@ -33,12 +42,50 @@
...
-  @Inject RemoteStorageManager remoteStorageManager;
 
-  @Inject @ApplicationContext Context context;
+  private final Context context;
+  private final RemoteStorageManager remoteStorageManager;
+  private final OfflineUuidGenerator uuidGenerator;
 
   @Inject
-  public UserMediaRepository() {}
+  public UserMediaRepository(
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/repository/UserMediaRepository.java:51]: You read my mind. Great!

Comment (gino-m) [reply gnd/src/main/java/com/google/android/gnd/repository/UserMediaRepository.java:51]: ..or at least you read my `TODO` ;)

File: gnd/src/main/java/com/google/android/gnd/system/CameraManager.java [hunk 2]
```diff
@@ -37,61 +41,63 @@
...
+      ActivityStreams activityStreams) {
+    this.context = context;
     this.permissionsManager = permissionsManager;
     this.activityStreams = activityStreams;
   }
 
   /** Launches the system's photo capture flow, first obtaining permissions if necessary. */
   @Cold
-  public Maybe<Bitmap> capturePhoto() {
+  public Maybe<Nil> capturePhoto(File destFile) {
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/system/CameraManager.java:60]: Would it make more sense for this and related methods to return `Completable<Boolean>`?

**Review by gino-m — CHANGES_REQUESTED** [review]
Great work, thanks for getting this done so quickly!! One question - is there a way to get the photos to appear in a folder called "Ground Photos" instead of "Pictures"? I tried upding the file_paths.xml and that didn't work. Any ideas?

**Comment by shobhitagarwal1612** [comment]
Pictures is the name of media authority which is hard coded. I haven't tried using a different one but could be possible. Would it be possible to address the remaining issues later? I've kept my laptop aside for the day. :)

**Comment by gino-m** [comment]
> Pictures is the name of media authority which is hard coded. I haven't tried using a different one but could be possible. Would it be possible to address the remaining issues later? I've kept my laptop aside for the day. :)

Certainly! I already built an apk and sent it for testing, but let's merge this one as-is anyway and address those secondary issues separately. Thanks again, great work!

**Review by gino-m — APPROVED** [review]

**Pull request merged**